src/backend/common/queries/team_query.py

```diff
@@ -98,6 +99,18 @@ def _query_async(
             DistrictTeam.district_key == ndb.Key(District, district_key)
         ).fetch_async()
         team_keys = map(lambda district_team: district_team.team, district_teams)
+
+        matches = re.match(r"\d{4,}", district_key)
+        if matches is not None:
+            year = int(matches.group(0))
+            event_teams = EventTeam.query(EventTeam.year == year).fetch()
```

Comment:
This query is pretty expensive. Using `.fetch(keys_only=True)` should be sufficient (although you'll need to parse out the team key from the EventTeam key).

Reply:
Also, the EventTeam query can probably be done asynchronously with `DistrictTeam.query` by using `fetch_async()`

e.g.
`district_teams, event_teams = yield DistrictTeam.query(...), EventTeam.query().fetch_async(keys_only=True)`